src/ior.c

```diff
@@ -374,36 +374,56 @@ static void CheckFileSize(IOR_test_t *test, char * testFilename, IOR_offset_t da
  * difference in buffers and returns total errors counted.
  */
 static size_t
-CompareBuffers(void *expectedBuffer,
-               void *unknownBuffer,
-               size_t size,
-               IOR_offset_t transferCount, IOR_param_t *test, int access)
+CompareData(void *expectedBuffer, size_t size, IOR_offset_t transferCount, IOR_param_t *test, IOR_offset_t offset, int fillrank, int access)
 {
         char testFileName[MAX_PATHLEN];
         char bufferLabel1[MAX_STR];
         char bufferLabel2[MAX_STR];
-        size_t i, j, length, first, last;
+        size_t i, j, length;
         size_t errorCount = 0;
-        int inError = 0;
-        unsigned long long *goodbuf = (unsigned long long *)expectedBuffer;
-        unsigned long long *testbuf = (unsigned long long *)unknownBuffer;
+
+        IOR_offset_t offsetSignature = 0;
+        unsigned long long hi, lo, val; // for data verification
+        hi = ((unsigned long long)fillrank) << 32;
+        lo = (unsigned long long)test->timeStampSignatureValue;
+        if (test->storeFileOffset){
+          offsetSignature = offset;
+        }
+
+        unsigned long long *testbuf = (unsigned long long *)expectedBuffer;
 
         if (access == WRITECHECK || access == READCHECK) {
                 strcpy(bufferLabel1, "Expected: ");
                 strcpy(bufferLabel2, "Actual:   ");
         } else {
-                ERR("incorrect argument for CompareBuffers()");
+                ERR("incorrect argument for CompareData()");
         }
 
         length = size / sizeof(IOR_size_t);
-        first = -1;
         if (verbose >= VERBOSE_3) {
                 fprintf(out_logfile,
                         "[%d] At file byte offset %lld, comparing %llu-byte transfer\n",
                         rank, (long long) offset, (long long)size);
         }
+
+        int incompressibleSeed = test->setTimeStampSignature + fillrank;
```

Comment:
This line does not seem intuitive at all.

Reply:
This is by necessity the same as used elsewhere in the code to generate the PRNG data (e.g. line 645).